lib/lf/base/comm.h

```diff
@@ -0,0 +1,238 @@
+
+extern void ParseCommandLine(const int argc = 0, const char** argv = nullptr);
+
+extern bool ParseFile(const std::string& file = "");
+
+/**
+ * @brief Add possible input for variable called `name` with description
+ *        `comment`.
+ * @param name The name of the variable.
+ * @param comment Description of the variable.
+ */
+template <typename T>
+void Add(const std::string& name, const std::string& comment) {
+  kDesc.add_options()(name.c_str(), po::value<T>(), comment.c_str());
+}
+
+/**
+ * @brief Add possible input for variable called `name` with description
+ *        `comment` with default value `def`.
+ * @param name The name of the variable.
+ * @param comment Description of the variable.
+ * @param def T The default value.
+ */
+template <typename T>
+void Add(const std::string& name, const std::string& comment, const T& def) {
+  kDesc.add_options()(name.c_str(), po::value<T>()->default_value(def),
+                      comment.c_str());
+}
+
+template <class A>
+std::function<void(unsigned int)> SetCtrl(A& a) {
+  std::function<void(unsigned int)> lambda = [&a](unsigned int c) {
+    a.output_ctrl_ = c;
+  };
+  return lambda;
+}
+
+/**
+ * @brief Possibility of setting the public member `ctrl` of the
+ *        `class_instance`. Option is called `name` with description `comment`
+ * @param name The name of the option.
+ * @param class_instance Instance of the class wher to change member value.
+ * @param comment (optional) Description of the option.
+ */
+template <class A>
+void AddCtrl(const std::string& name, A& class_instance,
+             const std::string& comment) {
+  // Why not solve it like:
+  //   void AddCtrl(const string& name, const T& val, const string& comment);
+  // Call it like:
+  //   AddCtrl("ctrl", MyClass.PublicVar, "Some comment");
+  //   AddCtrl("ctrl", MyClass::StaticVar, "Some comment");
+  // ?
+  kDesc.add_options()(
+      name.c_str(),
+      po::value<unsigned int>()->notifier(SetCtrl(class_instance)),
+      comment.c_str());
+}
+
+template <typename T>
+std::function<void(T)> SetValue(T& value) {
+  std::function<void(T)> lambda = [&value](T new_value) { value = new_value; };
+  return lambda;
+}
+
+/**
+ * @brief Possibility of setting any variable `value`. Option is called `name`
+          with description `comment`
+ * @param name The name of the option.
+ * @param value The variable to set.
+ * @param comment (optional) Description of the option.
+ */
+template <typename T>
+void AddSetter(const std::string& name, T& value, const std::string& comment) {
+  // Avoid duplicate entries
+  try {
+    // Don't add the option if it exists already (then no error is thrown)
+    // false -> only exact match in name is admissible
+    po::option_description el = kDesc.find(name, false);
+  } catch (const std::exception e) {
+    // If not found, then we add the option
+    kDesc.add_options()(name.c_str(), po::value<T>()->notifier(SetValue(value)),
```

Comment:
Why not declare the lambda directly here?
```
auto lambda = [&value](T new_value) { value = new_value; };
kDesc.add_options()(name.c_str(), po::value<T>()->notifier(lambda),
```

Reply:
Since I didn't the compiler to define `SetValue` as often as `AddSetter` is called. 
It's templated so it will probably be defined multiple times but not as frequently as it's the case if we put it inside `AddSetter`. 
Then again I don't know if that makes a difference perfomance wise at all. We can also put it inside `AddSetter` for reasons of readability and one namespace less (namely `lf::base::ci::internal`)

Reply:
I don't think that omitting SetValue would result in any performance penalty because
- SetValue is also a template function, so whenever AddSetter is instantiated with a new type `T`, a new instantiation of `SetValue` is also created.
- For sufficently high optimization levels, the compiler should anyway inline `SetValue` in `AddSetter`. Function calls are expensive and by inlining `SetValue` he can optimize the code a lot more. Of course the executable size increases but in my experience this is a very minor effect...

Reply:
Ok, I've put it inside the function `AddSetter`.